sklearn/tests/test_multioutput.py

```diff
@@ -588,3 +588,16 @@ def test_multioutput_estimator_with_fit_params(estimator, dataset):
     estimator.fit(X, y, some_param=some_param)
     for dummy_estimator in estimator.estimators_:
         assert 'some_param' in dummy_estimator._fit_params
+
+
+def test_regressor_chain_w_fit_params():
+    rng = np.random.RandomState(0)
+    X, y = datasets.make_regression(n_targets=3)
+    weight = rng.random((y.shape[0], ))
+
+    model = RegressorChain(SGDRegressor())
```

Comment:
We need to check that the underlying estimator received these weights.

This can be done by creating a mock estimator and when fit is called, store the `sample_weight` as an attribute.

Later after `RegressorChain.fit` is called, then we can assert that the `sample_weight` is the same as the is passed into `RegressorChain.fit`.

Reply:
@venkyyuvy I think that something like this would do the trick:


```py

class MySGD(SGDRegressor):

	def fit(X, y, **fit_params):
		self.sample_weight = fit_params['sample_weights]
		super().fit(X, y, fit_params)

# ...

model = RegressorChain(MySGD())
assert model.IDontRememberTheAttributeName.sample_weight is weight
```

Ideally we'd have a chain with 2 estimators (it can just be 2 MySGD)

Reply:
sure. Updated it now.